com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonReferenceProposalProvider.java

```diff
@@ -112,66 +125,103 @@ public String value() {
         public String label() {
             return label;
         }
-        
+
+        public boolean isLocalOnly() {
+            return isLocalOnly;
+        }
+
+        /**
+         * Returns all proposals found in the document at the specified field.
+         * 
+         * @param document
+         * @param fieldName
+         * @param path
+         * @return Collection of proposals
+         */
+        public Collection<Proposal> collectProposals(JsonNode document, IPath path) {
+            final Collection<Proposal> results = Lists.newArrayList();
+            if (value() == null) {
+                return results;
+            }
+
+            final JsonNode parameters = ValidationUtil.findNode(value(), document);
+            if (parameters == null) {
+                return results;
+            }
+
+            final String basePath = (path != null ? path.toString() : "") + "#/" + value() + "/";
+
+            for (Iterator<String> it = parameters.fieldNames(); it.hasNext();) {
+                String key = it.next();
+                String value = basePath + key.replaceAll("/", "~1");
+                String encoded = URLUtils.encodeURL(value);
+
+                results.add(new Proposal("\"" + encoded + "\"", key, null, value));
+            }
+
+            return results;
+        }
+
         public static ContextTypeCollection newContentTypeCollection(Iterable<ContextType> contextTypes) {
-        	return new ContextTypeCollection(contextTypes);
+            return new ContextTypeCollection(contextTypes);
+        }
+
+        public static ContextTypeCollection emptyContentTypeCollection() {
+            return new ContextTypeCollection(Collections.<ContextType> emptyList());
         }
-        
-		public static ContextTypeCollection emptyContentTypeCollection() {
-			return new ContextTypeCollection(Collections.<ContextType>emptyList());
-		}
-    }
-    
-    public static class ContextTypeCollection {
-    	
-    	private final Iterable<ContextType> contextTypes;
-
-		protected ContextTypeCollection(Iterable<ContextType> contextTypes) {
-			this.contextTypes = contextTypes;
-    	}
-		
-		public ContextType get(String path) {
-			if (Strings.emptyToNull(path) == null) {
-				return ContextType.UNKNOWN;
-			}
-			for (ContextType next : contextTypes) {
-				if (path.matches(next.regex)) {
-					return next;
-				}
-			}
-			return ContextType.UNKNOWN;
-		}
     }
 
-    /**
-     * Returns all proposals found in the document at the specified field.
-     * 
-     * @param document
-     * @param fieldName
-     * @param path
-     * @return Collection of proposals
-     */
-    public Collection<Proposal> collectProposals(JsonNode document, String fieldName, IPath path) {
-        final Collection<Proposal> results = Lists.newArrayList();
-        if (fieldName == null) {
-            return results;
+    public static class ContextTypeValue extends ContextType {
```

Comment:
Now, when we have two kinds of `ContextType`, it makes to rename the original to something like `ReferenceContextType` to make it clear how it differentiate from the `ValueContextType`. A Javadoc would also be helpful.